text/0015-binary-to-ternary-encoding/0015-binary-to-ternary-encoding.md

```diff
@@ -0,0 +1,79 @@
++ Feature name: `binary-to-ternary-encoding`
++ Start date: 2020-06-08
++ RFC PR: [iotaledger/protocol-rfcs#0015](https://github.com/iotaledger/protocol-rfcs/pull/15)
+
+# Summary
+
+In the IOTA protocol, a transaction is represented as ternary data. However, sometimes it is necessary to store binary data (e.g. the digest of a binary hash function) inside of a transaction. This requires the conversion of binary into ternary strings.
+The IOTA client libraries support the opposite conversion that encodes 5 trits as 1 byte (sometimes also referred to as `t5b1`), which is used for network communication and in storage layers. This RFC describes the corresponding counterpart to encode 1 byte as 6 trits.
+
+# Motivation
+
+A byte is composed of 8 bits that can represent 2<sup>8</sup> = 256 different values. On the other hand, 6 trits can hold 3<sup>6</sup> = 729 values while 5 trits can hold 3<sup>5</sup> = 243 values. Therefore, the most memory-efficient way to encode one byte requires the use of 6 trits. Although there exist many potential encoding schemes to convert binary data into ternary, the proposed version has been designed to directly match the widely used `t5b1` encoding.
+
+# Detailed design
+
+### Bytes to trits
+In order to encode a binary string S into ternary, each byte of S is interpreted as a signed (two's complement) 8-bit integer value v. Then, v is encoded as a little-endian 6-trit string in balanced ternary representation. Finally, the resulting groups of trits are concatenated.
+
+This algorithm can also be described using the following pseudocode:
+```
+T ← []
+foreach byte b in S:
+  v ← int8(b)
+  g ← IntToTrits(v, 6)
```

Comment:
So `IntToTrits` ( and`TritsToInt`) is a blackbox?

Just from reading the RFC I wouldn't be able to implement it...
Maybe you want the developer to depend on you examples?
If so can you make it clearer please?

Reply:
It is out of the scope of the protocol RFCs to formally define aspects of the current protocol. For this, the reader needs to refer to existing docs/spec outside of the repository.